cmd/imagePushToRegistry.go

```diff
@@ -191,27 +199,33 @@ func pushLocalImageToTargetRegistry(config *imagePushToRegistryOptions, utils im
 	}
 	log.Entry().Infof("Loading local image... Done")
 
-	for i := 0; i < len(config.TargetImages); i++ {
-		dst := fmt.Sprintf("%s/%s", config.TargetRegistryURL, config.TargetImages[i])
+	for _, trgImage := range config.TargetImages {
+		trgImage := trgImage
+		targetImage, ok := trgImage.(string)
+		if !ok {
+			return fmt.Errorf("incorrect name of target image: %v", trgImage)
+		}
 
-		g.Go(func() error {
-			log.Entry().Infof("Pushing %s...", dst)
-			if err := utils.PushImage(ctx, img, dst, platform); err != nil {
-				return err
-			}
-			log.Entry().Infof("Pushing %s... Done", dst)
-			return nil
-		})
+		if config.TargetImageTag != "" {
+			g.Go(func() error {
+				dst := fmt.Sprintf("%s/%s:%s", config.TargetRegistryURL, targetImage, config.TargetImageTag)
+				log.Entry().Infof("Pushing %s...", dst)
+				if err := utils.PushImage(ctx, img, dst, platform); err != nil {
+					return err
+				}
+				log.Entry().Infof("Pushing %s... Done", dst)
+				return nil
+			})
+		}
 
 		if config.TagLatest {
 			g.Go(func() error {
-				// imageName is repository + image, e.g test.registry/testImage
-				imageName := parseDockerImageName(dst)
-				log.Entry().Infof("Pushing %s...", imageName)
-				if err := utils.PushImage(ctx, img, imageName, platform); err != nil {
+				dst := fmt.Sprintf("%s/%s", config.TargetRegistryURL, targetImage)
+				log.Entry().Infof("Pushing %s...", dst)
+				if err := utils.PushImage(ctx, img, dst, platform); err != nil {
 					return err
 				}
-				log.Entry().Infof("Pushing %s... Done", imageName)
+				log.Entry().Infof("Pushing %s... Done", dst)
 				return nil
 			})
```

Comment:
the same as above

Reply:
https://github.com/SAP/jenkins-library/pull/4707/files#r1423440662